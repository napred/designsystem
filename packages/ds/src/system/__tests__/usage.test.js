/**
 * @flow
 */

import React from 'react';
import { css } from 'styled-components';
import { render } from 'enzyme';
import { createCssStyle, createStringStyle, createSystemComponent, DesignSystem } from '../';
import type { StyleFn } from '../types';

describe('design system usage', () => {
  it('works correctly', () => {
    const bgColor: StyleFn<{ bgColor?: string | void }> = createCssStyle(
      ['bgColor'],
      css`
          /* prettier-ignore */
          ${({ bgColor: color, system }) =>
            color ? `background-color: ${system.color(color, color)};` : ''}
        `,
    );
    const display: StyleFn<{ display?: string | void }> = createStringStyle(
      'display',
      'display',
      'block',
    );
    const opacity: StyleFn<{ opacity?: number | void }> = createStringStyle('opacity', 'opacity');

    const SpanOpacity = createSystemComponent('span', [opacity]);
    const SpanColorOpacity = createSystemComponent(SpanOpacity);
    const Div = createSystemComponent('div', [display]);
    const CssParagraph = createSystemComponent('p', [bgColor]);
    const BlackDiv = createSystemComponent('div');

    // $FlowExpectError
    BlackDiv.defaultProps = {
      bgColor: 'black',
    };

    const WhiteDiv = createSystemComponent(BlackDiv);
    const TomatoDiv = createSystemComponent(WhiteDiv);
    // $FlowExpectError
    TomatoDiv.defaultProps = {
      color: 'tomato',
    };

    const wrapper = render(
      <DesignSystem>
        <SpanOpacity />
        <SpanOpacity opacity={1} />
        <SpanColorOpacity />
        <SpanColorOpacity color="tomato" />
        <SpanColorOpacity color="tomato" opacity={1} />
        <Div />
        <Div as={SpanOpacity} display="flex" />
        <Div as={SpanColorOpacity} color="tomato" />
        <Div as={SpanColorOpacity} color="tomato" display="flex" />
        <Div as={SpanColorOpacity} color="tomato" opacity={0.5} />
        <SpanColorOpacity as="a" color="tomato" />
        <Div as="p" display="black" />
        <CssParagraph />
        <CssParagraph bgColor="tomato" />
        <CssParagraph bgColor="primary" />
        <CssParagraph as={Div} display="flex" bgColor="primary" />
        <BlackDiv />
        <BlackDiv bgColor="white" />
        <Div as={BlackDiv} />
        <Div as={BlackDiv} bgColor="white" />
        <WhiteDiv />
        <WhiteDiv bgColor="tomato" />
        <TomatoDiv />
        <TomatoDiv color="white" />
        <TomatoDiv bgColor="white" color="white" />
      </DesignSystem>,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
