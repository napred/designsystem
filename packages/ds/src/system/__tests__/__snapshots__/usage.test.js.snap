// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`design system usage works correctly 1`] = `
Array [
  .c0 {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
}

<span
    class="c0"
  />,
  .c0 {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  opacity: 1;
}

<span
    class="c0"
  />,
  .c0 {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
}

<span
    class="c0"
  />,
  .c0 {
  color: tomato;
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
}

<span
    class="c0"
  />,
  .c0 {
  opacity: 1;
  color: tomato;
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
}

<span
    class="c0"
  />,
  .c0 {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  display: block;
}

<div
    class="c0"
  />,
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<span
    class="c0"
  />,
  .c1 {
  color: tomato;
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
}

.c0 {
  color: tomato;
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  display: block;
}

<span
    class="c0 c1"
  />,
  .c1 {
  color: tomato;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
}

.c0 {
  color: tomato;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<span
    class="c0 c1"
  />,
  .c1 {
  color: tomato;
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  opacity: 0.5;
}

.c0 {
  color: tomato;
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  display: block;
}

<span
    class="c0 c1"
  />,
  .c0 {
  color: tomato;
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
}

<a
    class="c0"
  />,
  .c0 {
  display: black;
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  display: black;
}

<p
    class="c0"
  />,
  .c0 {
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
}

<p
    class="c0"
  />,
  .c0 {
  background-color: tomato;
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  background-color: tomato;
}

<p
    class="c0"
  />,
  .c0 {
  background-color: #D03C70;
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  background-color: #D03C70;
}

<p
    class="c0"
  />,
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #D03C70;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  background-color: #D03C70;
}

<div
    class="c0"
  />,
  .c0 {
  background-color: black;
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
}

<div
    class="c0"
  />,
  .c0 {
  background-color: #FFFFFF;
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
}

<div
    class="c0"
  />,
  .c0 {
  background-color: black;
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  display: block;
}

<div
    class="c0"
  />,
  .c0 {
  background-color: #FFFFFF;
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
  display: block;
}

<div
    class="c0"
  />,
  .c0 {
  background-color: black;
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
}

<div
    class="c0"
  />,
  .c0 {
  background-color: tomato;
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
}

<div
    class="c0"
  />,
  .c1 {
  background-color: black;
  color: tomato;
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
}

.c0 {
  color: tomato;
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
}

<div
    class="c0 c1"
  />,
  .c1 {
  background-color: black;
  color: #FFFFFF;
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
}

.c0 {
  color: #FFFFFF;
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
}

<div
    class="c0 c1"
  />,
  .c1 {
  background-color: #FFFFFF;
  color: #FFFFFF;
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
}

.c0 {
  background-color: #FFFFFF;
  color: #FFFFFF;
  font-family: "Palanquin","Helvetica",sans-serif;
  font-size: 16px;
}

<div
    class="c0 c1"
  />,
]
`;
